src/modules/acbs/acbs-facility-fixed-fee.service.test.ts

```diff
@@ -0,0 +1,112 @@
+import { HttpService } from '@nestjs/axios';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { AxiosError } from 'axios';
+import { when } from 'jest-when';
+import { of, throwError } from 'rxjs';
+
+import { AcbsFacilityFixedFeeService } from './acbs-facility-fixed-fee.service';
+import { AcbsGetFacilityFixedFeeResponseDto, AcbsGetFacilityFixedFeeResponseItem } from './dto/acbs-get-facility-fixed-fee-response.dto';
+import { AcbsException } from './exception/acbs.exception';
+
+describe('AcbsFacilityFixedFeeService', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const idToken = valueGenerator.string();
+  const baseUrl = valueGenerator.httpsUrl();
+  const portfolioIdentifier = valueGenerator.string({ length: 2 });
+  const facilityIdentifier = valueGenerator.facilityId();
+
+  let httpService: HttpService;
+  let service: AcbsFacilityFixedFeeService;
+
+  let httpServiceGet: jest.Mock;
+
+  const expectedHttpServiceGetArgs = [
+    `/Portfolio/${portfolioIdentifier}/Facility/${facilityIdentifier}/Fee`,
+    {
+      baseURL: baseUrl,
+      headers: { Authorization: `Bearer ${idToken}` },
+    },
+  ];
+
+  beforeEach(() => {
+    httpService = new HttpService();
+
+    httpServiceGet = jest.fn();
+    httpService.get = httpServiceGet;
+
+    service = new AcbsFacilityFixedFeeService({ baseUrl }, httpService);
+  });
+
+  const generateFacilityFixedFee = (): AcbsGetFacilityFixedFeeResponseItem => ({
+    FixedFeeAmount: valueGenerator.nonnegativeFloat({ fixed: 2 }),
+    EffectiveDate: valueGenerator.dateTimeString(),
+    ExpirationDate: valueGenerator.dateTimeString(),
+    NextDueDate: valueGenerator.dateTimeString(),
+    NextAccrueToDate: valueGenerator.dateTimeString(),
+    SegmentIdentifier: valueGenerator.stringOfNumericCharacters({ length: 2 }),
+    Description: valueGenerator.string(),
+    Currency: {
+      CurrencyCode: valueGenerator.string(),
+    },
+    LenderType: {
+      LenderTypeCode: valueGenerator.stringOfNumericCharacters({ length: 3 }),
+    },
+    IncomeClass: {
+      IncomeClassCode: valueGenerator.string({ length: 3 }),
+    },
+    SpreadToInvestorsIndicator: valueGenerator.boolean(),
+  });
+
+  const facilityFixedFeesInAcbs: AcbsGetFacilityFixedFeeResponseDto = [generateFacilityFixedFee(), generateFacilityFixedFee()];
+
+  describe('getFixedFeesForFacility', () => {
+    it('returns the fixed fees for the facility from ACBS if ACBS responds with the fixed fees', async () => {
+      when(httpServiceGet)
```

Comment:
Any reason why would this endpoint will return an empty array?
I would expect `404` instead.

Current implementation returns 404 when no fixed fee exists.
![image](https://user-images.githubusercontent.com/32125108/235122696-d2b6bce0-c53d-4aef-b778-bd9fd726282c.png)

Reply:
As per the ticket spec and acceptance criteria - I'll have a bit more of an investigation as to the existing status quo after lunch

Reply:
I've just confirmed that ACBS is not returning a 404, rather a 200 in the case of facility not found. The removal of 404 is part of the approach of this ticket as we cannot say whether the facility does not exist or there are just no fixed fees -- Happy to talk about this further if needed!

Reply:
Discussed to keep it to `200` with an empty array.